pkg/test/integration/common/framework.go

```diff
@@ -0,0 +1,837 @@
+				gomega.Eventually(c.TargetCluster.GetNumberOfNodes, 180, 5).Should(gomega.BeNumerically("==", initialNodes+3))
+				gomega.Eventually(c.TargetCluster.GetNumberOfReadyNodes, 180, 5).Should(gomega.BeNumerically("==", initialNodes+3))
+			})
+		})
+		ginkgo.Context("scale-up with replicas=6", func() {
+			ginkgo.It("should not lead to errors and add futher 3 nodes to target cluster", func() {
+				retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					machineDployment, _ := c.ControlCluster.McmClient.MachineV1alpha1().MachineDeployments(controlClusterNamespace).Get("test-machine-deployment", metav1.GetOptions{})
+					machineDployment.Spec.Replicas = 6
+					_, updateErr := c.ControlCluster.McmClient.MachineV1alpha1().MachineDeployments(controlClusterNamespace).Update(machineDployment)
+					return updateErr
+				})
+				ginkgo.By("Checking for errors")
+				gomega.Expect(retryErr).NotTo(gomega.HaveOccurred())
+				ginkgo.By("Checking number of ready nodes are 6 more than initial")
+				gomega.Eventually(c.TargetCluster.GetNumberOfNodes, 180, 5).Should(gomega.BeNumerically("==", initialNodes+6))
+				gomega.Eventually(c.TargetCluster.GetNumberOfReadyNodes, 180, 5).Should(gomega.BeNumerically("==", initialNodes+6))
+			})
+
+		})
+		ginkgo.Context("scale-down with replicas=2", func() {
+			ginkgo.It("should not lead to errors and remove 4 nodes from target cluster", func() {
+				retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					machineDployment, _ := c.ControlCluster.McmClient.MachineV1alpha1().MachineDeployments(controlClusterNamespace).Get("test-machine-deployment", metav1.GetOptions{})
+					machineDployment.Spec.Replicas = 2
+					_, updateErr := c.ControlCluster.McmClient.MachineV1alpha1().MachineDeployments(controlClusterNamespace).Update(machineDployment)
+					return updateErr
+				})
+				ginkgo.By("Checking for errors")
+				gomega.Expect(retryErr).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("Checking number of ready nodes are 2 more than initial")
+				gomega.Eventually(c.TargetCluster.GetNumberOfNodes, 300, 5).Should(gomega.BeNumerically("==", initialNodes+2))
+				gomega.Eventually(c.TargetCluster.GetNumberOfReadyNodes, 300, 5).Should(gomega.BeNumerically("==", initialNodes+2))
+			})
+			// rapid scaling back to 2, should lead to freezing and unfreezing
+			ginkgo.It("should freeze and unfreeze machineset temporarily", func() {
+				if mcsession == nil {
+					// controllers running in pod
+					// Create log file from container log
+					mcmOutputFile, err := rotateLogFile(mcmLogFile)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					mcOutputFile, err := rotateLogFile(mcLogFile)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					ginkgo.By("Reading container log is leading to no errors")
+					podList, err := c.ControlCluster.Clientset.CoreV1().Pods(controlClusterNamespace).List(metav1.ListOptions{
+						LabelSelector: "app=machine-controller-manager",
```

Comment:
Hi @rkjakeer,

This is one last comment. The Label selector must be "role" instead of "app" because, this is how MCM deployment looks like

```
"labels": {
    "app": "kubernetes",
    "role": "machine-controller-manager"
  }
```

```suggestion
						LabelSelector: "role=machine-controller-manager",
```

Reply:
This is done. Thanks 👍